plugins/modules/api.py

```diff
@@ -282,11 +282,7 @@ def main():
     )
 
     try:
-        client = Client(
-            host=module.params["cluster_instance"]["host"],
-            username=module.params["cluster_instance"]["username"],
-            password=module.params["cluster_instance"]["password"],
-        )
+        client = Client.get_client(module.params["cluster_instance"])
```

Comment:
Made this into a `classmethod` in every module ... seems more appropriate since it's creating an instance of a class anyways.